src/util.cc

```diff
@@ -1652,14 +1684,47 @@ int choose_cpu(BindCPU bind_cpu) {
   // presumably due to cheaper context switching and/or
   // better interaction with CPU frequency scaling.
   if (bind_cpu >= 0) {
+    if (cpu_lock_fd_out.is_open()) {
+      struct flock lock {
+        .l_type = F_WRLCK,
+        .l_whence = SEEK_SET,
+        .l_start = bind_cpu,
+        .l_len = 1,
+        .l_pid = 0
+      };
+      // Try to acquire the lock for this CPU
+      (void)fcntl(cpu_lock_fd_out, F_SETLK, &lock);
+      // Ignore fcntl errors - nothing we can do
```

Comment:
If we do two rr recordings manually bound to the same CPU, and then one finishes, I guess the lock is released and another not-manually bound rr could bind to that CPU. Nothing to worry about in practice I suppose.

Reply:
Yes, that's true. What if the explicitly bound processes acquired read locks, and the implicit ones acquired write locks? I think that would solve this situation?

Reply:
Let's just not care.

Reply:
Ok by me. One thing to point out though is that the same applies with replays, which are more likely to be bound, since we record which CPU they go on in the trace.

Reply:
We're only calling `choose_cpu` if we failed to bind to the CPU in the trace, right? So most replays won't do any locking at all.

But I guess that creates a problem of its own: when an rr replay runs bound to a CPU, it's not locked so an rr recording might choose that CPU.

Reply:
Ah, fair point. I thought that case would be covered, but you're right it's not. We should definitely be locking that CPU in the replay, so we should always have replay call choose_cpu with the index from the trace.